### PR TITLE
🗃️ remove nullable constraint for course credit_hours

### PR DESCRIPTION
**Pull request by vikaspotluri123**
refs gradebook/client#420

Also a couple more QoL improvements - review by-commit.

### CODE REVIEW

File: lib/database/migration-utils/database-type.js [hunk 1]
```diff
@@ -0,0 +1,8 @@
+// @ts-check
+
+/**
+ * @param {import('knex').Knex} knex
+ */
+export function getDatabaseType(knex) {
+	return knex.client.config.client === 'sqlite3' ? 'sqlite3' : 'mysql';
```
Comment (ramsays) [inline lib/database/migration-utils/database-type.js:7]: We love a good common function 👍

File: test/fixtures/example-data.js [hunk 2]
```diff
@@ -5,11 +5,11 @@ const fixtures = [
...
-	['courses', {id: '5d3c81fbaf84df3df0a18c1b', user_id: '5d3c8150930db248a9d7514f', semester: '2019S', name: 'CORS 101', cutoffs: '{"A":90,"B":80,"C":70,"D":60}', credit_hours: null, settings: '{"alias": "Course A"}'}],
-	['courses', {id: '5d3c8227faade0e6c7e0ed30', user_id: '5d3c8150930db248a9d7514f', semester: '2000S', name: 'CORS 202', cutoffs: '{"A":90,"B":80,"C":70,"D":60}', credit_hours: null}],
-	['courses', {id: '5d3c822b361ac86457c4395e', user_id: '5d3c8150930db248a9d7514f', semester: '2019S', name: 'CORS 303', cutoffs: '{"A":90,"B":80,"C":70,"D":60}', credit_hours: null}],
-	['courses', {id: '5d3c8230c95745a3e0806c9b', user_id: '5d3c8150930db248a9d7514f', semester: '2019S', name: 'CORS 404', cutoffs: '{"A":90,"B":80,"C":70,"D":60}', credit_hours: null}],
-	['courses', {id: '5d48437d8c01946f39ea997f', user_id: '5d3c8150930db248a9d7514f', semester: '2019S', name: 'GRDS 000', cutoffs: '{"A":90,"B":80,"C":70,"D":60}', credit_hours: null}],
+	['courses', {id: '5d3c81fbaf84df3df0a18c1b', user_id: '5d3c8150930db248a9d7514f', semester: '2019S', name: 'CORS 101', cutoffs: '{"A":90,"B":80,"C":70,"D":60}', credit_hours: 3, settings: '{"alias": "Course A"}'}],
```
Comment (ramsays) [inline test/fixtures/example-data.js:8]: Do we have any behavior in server that is specific to null credit courses? I wouldn't think so as that is a client function but just want to make sure.

File: lib/database/migrations/000009-drop-nullable-course-credits.js [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+const WITH_NULLABLE_SCHEMA = {
+	type: 'tinyint', nullable: true, fallback: null, unsigned: true, validations: {between: [0, MAX_CREDITS_IN_COURSE]},
+};
+
+/** @type {import('../schema.d').ColumnSchema} */
+const WITHOUT_NULLABLE_SCHEMA = {
+	type: 'tinyint', nullable: false, unsigned: true, validations: {between: [0, MAX_CREDITS_IN_COURSE]},
+};
+
+const {up, down} = createAlterColumnMigration({
```
Comment (ramsays) [inline lib/database/migrations/000009-drop-nullable-course-credits.js:15]: Is there a reason we have this as two different migrations?

File: test/fixtures/example-data.js (hunk 2 above)
Comment (vikaspotluri123) [reply test/fixtures/example-data.js:8]: I think it's just validation, but none of our tests that use these fixtures seem to need it!

File: lib/database/migrations/000009-drop-nullable-course-credits.js (hunk 3 above)
Comment (vikaspotluri123) [reply lib/database/migrations/000009-drop-nullable-course-credits.js:15]: Yes, it's good practice to only change either the data or the schema in a single migration

**Review by ramsays — APPROVED** [review]

**Review by joshcosta — APPROVED** [review]

**Pull request merged**